tests/PHPStan/Analyser/nsrt/pow.php

```diff
@@ -47,11 +47,11 @@ function (): void {
 		$x = 4;
 	}
 
-	assertType('int<4, 27>|int<16, 81>', pow($range, $x));
-	assertType('int<4, 27>|int<16, 81>', $range ** $x);
+	assertType('int<4, 81>', pow($range, $x));
+	assertType('int<4, 81>', $range ** $x);
 
-	assertType('int<4, 27>|int<16, 64>', pow($x, $range));
-	assertType('int<4, 27>|int<16, 64>', $x ** $range);
+	assertType('int<4, 64>', pow($x, $range));
+	assertType('int<4, 64>', $x ** $range);
```

Comment:
This is a result of `UnionType::getSortedTypes` being called. I haven't looked into it, but I guess it's a bug in `TypeCombinator::union`, because the result of a union shouldn't depend on the order of parameters.